spec/app/ics-030-middleware/BEST_PRACTICES.md

```diff
@@ -0,0 +1,71 @@
+# Best Practices When Designing Middleware
```

Comment:
Does the spec repo have any style conventions? We use the [google docs style convention](https://developers.google.com/style) on ibc-go which states only the first word in a [title](https://developers.google.com/style/capitalization) is capitalized